File: app/src/main/java/com/example/myapp/DBService/Location.java
package com.example.myapp.DBService;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "locations")
public class Location {
    @PrimaryKey
    @ColumnInfo(name = "id", index = true)
    public long id;
    @ColumnInfo(name = "name")
    public String name;
    @ColumnInfo(name = "country")
    public String country;
    @ColumnInfo(name = "coord_lon")
    public float lon;
    @ColumnInfo(name = "coord_lat")
    public float lat;
    @ColumnInfo(name = "favorite")
    boolean favorite;

    public Location(long id, String name, String country, float lon, float lat) {
        this.id = id;
        this.name = name;
        this.country = country;
        this.lon = lon;
        this.lat = lat;
        this.favorite = false;
    }
}

File: app/src/main/java/com/chplalex/shaman/DBService/Location.kt
package com.chplalex.shaman.DBService

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "locations")
class Location(
    @field:PrimaryKey
    @field:ColumnInfo(name = "id", index = true) var id: Long,
    @field:ColumnInfo(name = "name") var name: String,
    @field:ColumnInfo(name = "country") var country: String,
    @field:ColumnInfo(name = "coord_lon") var lon: Float,
    @field:ColumnInfo(name = "coord_lat") var lat: Float
) {
    @ColumnInfo(name = "favorite")
    var favorite = false
}